templates/base_index.html

```diff
@@ -1409,55 +1409,100 @@ <h2>
-        // Otherwise, randomly choose one of the takeovers and store it for next time
-        selectedIndex = Math.floor(Math.random() * selectedTakeovers.length);
+        }
       }
+    };
 
-      // Store the current takeover
-      localStorage.setItem('selected_takeover_index', selectedIndex)
+    xhr.open("GET", "/takeovers.json", true);
+    xhr.send();
 
-      // Get the takeover element
-      var selectedTakeover = selectedTakeovers[selectedIndex]
+  }
 
-      if (! document.body.contains(selectedTakeover)) {
-        // If it's not in the current document (as in, it's in a template) import it
-        // Note: This is for all browser *except* IE, which doesn't support <template>s
-        selectedTakeover = document.importNode(selectedTakeover, true);
-      }
+  function showTakeover(takeovers, index) {
+    // Default parameter
+    index = typeof index !== 'undefined' ? index : 0;
+
+    // Get HTML elements
+    var takeover = document.getElementById("takeover");
+    var title = document.getElementById("takeover-title");
+    var subtitle = document.getElementById("takeover-subtitle");
+    var image = document.getElementById("takeover-image");
+    var primaryUrl = document.getElementById("takeover-primary-url");
+    var secondaryUrl = document.getElementById("takeover-secondary-url");
+    var get_ctas = document.getElementById("takeover-ctas").getElementsByTagName("p")[0];
+    var takeoverAnimation = document.getElementsByClassName("p-takeover-animation")
 
-      // Replace the base takeover with the new one
-      // Note: We can't use Node.replaceWith (which would be nice), as it's not supported in IE
-      baseTakeover.parentNode.replaceChild(selectedTakeover, baseTakeover)
+    // Set values to homepage takeover
+    var selectedTakeover = takeovers[index];
+
+    takeover.className = "";
+    takeover.removeAttribute("style");
+
+    // Add takeover classes
+    var classNameString = "js-takeover p-takeover--" + selectedTakeover.class;
+    takeover.className += classNameString;
+
+    // Set language attributes
+    takeover.setAttribute("lang", selectedTakeover.lang);
+    takeover.setAttribute("lang-skip", selectedTakeover.lang_skip);
+
+    // Set takeover content
+    title.textContent = selectedTakeover.title;
+    subtitle.textContent = selectedTakeover.subtitle;
+    image.src = selectedTakeover.image;
+    image.removeAttribute("style");
+    image.style.width = selectedTakeover.image_width;
+    image.style.height = selectedTakeover.image_height;
+    primaryUrl.href = selectedTakeover.primary_url;
+    primaryUrl.textContent = selectedTakeover.primary_cta;
+    if (selectedTakeover.secondary_url && selectedTakeover.secondary_url !== "") {
+      secondaryUrl.href = selectedTakeover.secondary_url;
+      secondaryUrl.textContent = selectedTakeover.secondary_cta + " ›";
+    } else {
+      secondaryUrl.remove();
     }
+
+    // Remove animation delay
+    takeoverAnimation.className +=  ".is-loaded";
```

Comment:
You get an array back from `getElementsByClassName` therefore you need:
```js
takeoverAnimation[0].className += " is-loaded";
```
or switch to using an id.

Reply:
There is also `element.querySelectorAll()` that returns all child elements for given selector or `element.querySelector()` that returns first child element of given selector.

Reply:
Done. Additionally, I have added a fixed height (reduced a little the default takeover height), and centered aligned with flexbox. Tested on IE11, Opera 65 and Firefox 76 no major issues.

I'd rather avoid using query selectors, because you don't explicitly name them with a class or id. I would use id or class.